### ISSUE TITLE
File not found warning immediately replaced when key file not found

### ISSUE THREAD

**Issue by kortschak** [keepassxreboot/keepassxc#741]
When a key file is not found at the path specified, a less than helpful dialogue is shown:
```
Unable to open the database.
Wrong key or database file is corrupt.
```

## Expected Behavior

A dialogue showing that the key file was not found.
```
Can't open key file:
No such file or directory
```

## Current Behavior

Currently the dialogue that should be shown is flashed for about a quarter of a second and then replaced with:
```
Unable to open the database.
Wrong key or database file is corrupt.
```
It is very easy to miss the actual cause of the failure because of this.

## Possible Solution

The first error (file not found) should trump the second. This suggests to me that an ENOENT error condition is being ignored and keepassxc is forging ahead and failing again.

## Steps to Reproduce (for bugs)

1. Create a database with a password and key file.
2. Attempt to open the database with a bad path to the key file.

## Context

Was looking into replacing keepassx for the reasons that this fork exists.

## Debug Info

KeePassXC - 2.2.0 / 7580c385da58329ef29e0d026b1ac0720cbcccf2
Revision: caa49a8ef3ee28ed478192389b21d61107b3b8e0  / 7580c385da58329ef29e0d026b1ac0720cbcccf2

Libraries:
- Qt 5.5.1
- libgcrypt 1.6.5

Operating system: Ubuntu 16.04.2 LTS
CPU architecture: x86_64
Kernel: linux 4.4.0-81-generic

Enabled extensions:
- Auto-Type

**Comment by kortschak** [comment]
Looking at gui/DatabaseOpenWidget.cpp it looks like [`DatabaseOpenWidget::openDatabase` accepts](https://github.com/keepassxreboot/keepassxc/blob/7580c385da58329ef29e0d026b1ac0720cbcccf2/src/gui/DatabaseOpenWidget.cpp#L142) the [invalid `CompositeKey` returned by `DatabaseWidget::databaseKey` on failure](https://github.com/keepassxreboot/keepassxc/blob/7580c385da58329ef29e0d026b1ac0720cbcccf2/src/gui/DatabaseOpenWidget.cpp#L188). This probably shouldn't happen.

This fixes the problem:

```diff
diff --git a/src/gui/DatabaseOpenWidget.cpp b/src/gui/DatabaseOpenWidget.cpp
index 18b4b2b..60c450c 100644
--- a/src/gui/DatabaseOpenWidget.cpp
+++ b/src/gui/DatabaseOpenWidget.cpp
@@ -140,6 +140,8 @@ void DatabaseOpenWidget::openDatabase()
 {
     KeePass2Reader reader;
     CompositeKey masterKey = databaseKey();
+       if (masterKey.isEmpty())
+               return;
 
     QFile file(m_filename);
     if (!file.open(QIODevice::ReadOnly)) {
```

If this is an acceptable change I am happy to send a PR.

**Comment by louib** [comment]
@kortschak this should be fixed in `2.2.2`
